mgmtsystem/virtualcenter.py

```diff
@@ -131,58 +127,142 @@ def __init__(self, hostname, username, password, **kwargs):
         self.hostname = hostname
         self.username = username
         self.password = password
-        self._api = None
+        self._service_instance = None
+        self._content = None
         self._vm_cache = {}
         self.kwargs = kwargs
 
+    def __del__(self):
+        """Disconnect from the API when the object is deleted"""
+        if self._service_instance:
+            Disconnect(self._service_instance)
```

Comment:
In the context of how this code is used, a context manager would not help (it would cause multiple connections and disconnections to and from the API, and it would require refactoring of the code in `integration_tests` that uses this library. I followed the documentation for pyVmomi in calling disconnect when finished with the connection to the API. The current `psphere`-based code doesn't even bother to try to disconnect from the API. I know this isn't the ideal way to deal with this, but I don't really see a better option.